src/core/database.cpp

```diff
@@ -1193,6 +1196,66 @@ void Database::EndBatchSelect( ) {
     in_batch_select = false;
 }
 
+template <typename T>
+void Database::ProcessBatchSelectElement(T* ptr, int& argument_counter) {
+
+    if constexpr ( std::is_same_v<T, wxString> ) {
+        ptr[0] = sqlite3_column_text(batch_statement, argument_counter);
+    }
+    else if constexpr ( std::is_same_v<T, wxFileName> ) {
+        ptr[0] = sqlite3_column_text(batch_statement, argument_counter);
+    }
+    else if constexpr ( std::is_same_v<T, double> ) {
+        ptr[0] = sqlite3_column_double(batch_statement, argument_counter);
+    }
+    else if constexpr ( std::is_same_v<T, int> ) {
+        ptr[0] = sqlite3_column_int(batch_statement, argument_counter);
+    }
+    else if constexpr ( std::is_same_v<T, float> ) {
+        double temp_double = sqlite3_column_double(batch_statement, argument_counter);
+        ptr[0]             = float(temp_double);
+    }
+    else if constexpr ( std::is_same_v<T, long> ) {
+        ptr[0] = sqlite3_column_int64(batch_statement, argument_counter);
+    }
+    else {
+        MyPrintWithDetails("Error: Unknown format character!\n");
+    }
+    argument_counter++;
+}
+
+/**
+ * @brief This function takes a variable number of arguments, that are all pointers to data that
+ * will be fetched from the database. Rather than passing a char* literal to decode the types,
+ * we use a fold expresssion to call the ProcessBatchSelectElement function for each argument.
+ * The type of each pointer is checked there and the appropriate sqlite3 function is called, incrementing
+ * the argument counter.
+ * 
+ * @tparam Args 
+ * @param args 
+ * @return true 
+ * @return false 
+ */
+template <class... Args>
+bool Database::GetFromBatchSelect_NoChar(Args... args) {
```

Comment:
@jojoelfe this is kinda cool and might be useful in other places where we have to decode a string literal/const char* to deduce types. Just pointing it out for your curiousity.

Reply:
This is nice... Will it fail at compile time if an unsupported type is requested?

Reply:
As it was written, no. Branches are only thrown out after it is confirmed they are not evaluated, so the else clause would still just print a warning at runtime.

You might think you can just stick a static_assert(false, "message") in the else block, but that would always trigger, as all if constexpr() needs to be evaluated at compile time.

There is a little trick (I added here) that will instantiate the static_ProcessBatchSelectElement_no_match( ) template ONLY if the else block is reached, and then it takes the default template arg and throws the assert.

A bit confusing, but it works okay. It also makes for easier compiler error messages.